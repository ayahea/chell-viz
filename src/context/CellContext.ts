import { CELL_TYPE } from 'chell';
import * as React from 'react';

export const initialCellContext = {
  addCells: (cells: CELL_TYPE[]) => {
    return;
  },
  currentCells: [] as CELL_TYPE[],
  removeAllCells: (cells: CELL_TYPE[]) => {
    return;
  },
  removeCells: (cells: CELL_TYPE[]) => {
    return;
  },
};

export const CellContext = React.createContext({
  ...initialCellContext,
});
