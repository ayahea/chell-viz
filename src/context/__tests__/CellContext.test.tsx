import * as React from 'react';
import * as Renderer from 'react-test-renderer';

import CellContext, { initialCellContext } from '../CellContext';

describe('CellContext', () => {
  it('Should match the default snapshot.', () => {
    const wrapper = Renderer.create(
      <CellContext.Consumer>{context => React.createElement('div', context)}</CellContext.Consumer>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
    expect(wrapper.root.props).toEqual(initialCellContext);
  });

  it('Should have empty initial functions that produce 0 side effects.', () => {
    const wrapper = Renderer.create(
      <CellContext.Consumer>{context => React.createElement('div', context)}</CellContext.Consumer>,
    );
    wrapper.root.props.addCells();
    wrapper.root.props.removeAllCells();
    wrapper.root.props.removeCells();
    expect(wrapper.root.props).toEqual(initialCellContext);
  });
});
