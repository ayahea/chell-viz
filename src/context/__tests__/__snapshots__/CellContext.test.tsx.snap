// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellContext Should match the default snapshot. 1`] = `
<div
  addCells={[Function]}
  currentCells={Array []}
  removeAllCells={[Function]}
  removeCells={[Function]}
/>
`;
